Specify pull request title conventions :man_judge: (#564)
quantumblacklabs/kedro: CONTRIBUTING.md

| `fix`            | Non-breaking change which fixes an issue                                |
| `tests`          | Changes to project unit `tests/` and / or integration `features/` tests |

### Pull request title conventions

> Kedro requires that you [squash and merge your pull request commits](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#squash-and-merge-your-pull-request-commits). In most cases, the [merge message for a squash merge](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#merge-message-for-a-squash-merge) will default to the pull request title; therefore, pull request titles should follow [the rules pertaining to the subject of a great Git commit message](https://chris.beams.io/posts/git-commit/#seven-rules):

Reviewer: ```suggestion
The Kedro team ask that you [squash and merge your pull request commits](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#squash-and-merge-your-pull-request-commits) and, in most cases, the [merge message for a squash merge](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#merge-message-for-a-squash-merge) then defaults to the pull request title. 

For clarity, your pull request title should be descriptive and we ask you to follow some guidelines suggested by Chris Beams in his post [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/#seven-rules). In particular:
```

I'm not sure, but I don't think that Kedro (the tool) requires that you squash and merge? The Kedro repo? The Kedro team perhaps?

Reviewer: I've made a suggestion, see what you think and maybe it needs a further tweak. For example, perhaps we could link to Chris Beam's github account under his name. I'll leave it to you :)

Author: @stichbury I'm not an admin, but I believe the repo is set to require the "Squash and merge" option. At the very least, most contributors wouldn't have the. option to merge themselves, anyway. I provided this background more to inform why the PR title matters (i.e. because it's used as the commit message upon squash-merge.

Taking most of your suggestions:

```suggestion
The Kedro repository requires that you [squash and merge your pull request commits](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#squash-and-merge-your-pull-request-commits), and, in most cases, the [merge message for a squash merge](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#merge-message-for-a-squash-merge) then defaults to the pull request title.

For clarity, your pull request title should be descriptive and we ask you to follow some guidelines suggested by [Chris Beams](https://github.com/cbeams) in his post [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/#seven-rules). In particular:
```

* [Limit the pull request title to 50 characters](https://chris.beams.io/posts/git-commit/#limit-50)
* [Capitalize the pull request title](https://chris.beams.io/posts/git-commit/#capitalize)
* [Do not end the pull request title with a period](https://chris.beams.io/posts/git-commit/#end)

> * [Use the imperative mood in the pull request title](https://chris.beams.io/posts/git-commit/#imperative)

Reviewer: ```suggestion
* [Use the imperative tense for your pull request title](https://chris.beams.io/posts/git-commit/#imperative)
```

Reviewer: I think we should change to "tense" since that's correct. Imperative isn't a 'mood' as far as I know (though it sometimes seems that way!). I'll suggest an alternative for the bullets...

Reviewer: ```suggestion
For clarity, your pull request title should be descriptive and we ask you to follow some guidelines suggested by [Chris Beams](https://github.com/cbeams) in his post [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/#seven-rules). In particular, for your pull request title, we suggest that you do the following:

* [Limit the length to 50 characters](https://chris.beams.io/posts/git-commit/#limit-50)
* [Capitalise the first letter of the word](https://chris.beams.io/posts/git-commit/#capitalize)
* [Omit the period at the end](https://chris.beams.io/posts/git-commit/#end)
* [Use the imperative tense](https://chris.beams.io/posts/git-commit/#imperative)
```

Reviewer: I've taken out the duplication in the bullets. See what you think -- I'm not obsessive about this or anything, so go with what you like the most 😆

Author: ```suggestion
For clarity, your pull request title should be descriptive and we ask you to follow some guidelines suggested by [Chris Beams](https://github.com/cbeams) in his post [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/#seven-rules). In particular, for your pull request title, we suggest that you:

* [Limit the length to 50 characters](https://chris.beams.io/posts/git-commit/#limit-50)
* [Capitalise the first letter of the first word](https://chris.beams.io/posts/git-commit/#capitalize)
* [Omit the period at the end](https://chris.beams.io/posts/git-commit/#end)
* [Use the imperative tense](https://chris.beams.io/posts/git-commit/#imperative)
```
Minor adjustments. :)

| `fix`            | Non-breaking change which fixes an issue                                |
| `tests`          | Changes to project unit `tests/` and / or integration `features/` tests |

### Pull request title conventions

The Kedro repository requires that you [squash and merge your pull request commits](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#squash-and-merge-your-pull-request-commits), and, in most cases, the [merge message for a squash merge](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#merge-message-for-a-squash-merge) then defaults to the pull request title.

> For clarity, your pull request title should be descriptive and we ask you to follow some guidelines suggested by [Chris Beams](https://github.com/cbeams) in his post [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/#seven-rules). In particular, for your pull request title, we suggest that you:

Author: ```suggestion
For clarity, your pull request title should be descriptive, and we ask you to follow some guidelines suggested by [Chris Beams](https://github.com/cbeams) in his post [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/#seven-rules). In particular, for your pull request title, we suggest that you:
```

| `fix`            | Non-breaking change which fixes an issue                                |
| `tests`          | Changes to project unit `tests/` and / or integration `features/` tests |

### Pull request title conventions

The Kedro repository requires that you [squash and merge your pull request commits](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#squash-and-merge-your-pull-request-commits), and, in most cases, the [merge message for a squash merge](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#merge-message-for-a-squash-merge) then defaults to the pull request title.

> For clarity, your pull request title should be descriptive and we ask you to follow some guidelines suggested by [Chris Beams](https://github.com/cbeams) in his post [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/#seven-rules). In particular:
> 
> * [Limit the pull request title to 50 characters](https://chris.beams.io/posts/git-commit/#limit-50)
> * [Capitalize the pull request title](https://chris.beams.io/posts/git-commit/#capitalize)
> * [Do not end the pull request title with a period](https://chris.beams.io/posts/git-commit/#end)

Author: ```suggestion
```
